Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -1079,6 +1079,80 @@ class LengthVariantPropertyWrapper final : public PropertyWrapperGetter<const T&
     void (RenderStyle::*m_setter)(T&&);
 };
 
+class OptionalLengthPropertyWrapper : public PropertyWrapperGetter<std::optional<Length>> {
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    enum class Flags {
+        IsLengthPercentage = 1 << 0,
+        NegativeLengthsAreInvalid = 1 << 1,
+    };
+    OptionalLengthPropertyWrapper(CSSPropertyID property, std::optional<Length> (RenderStyle::*getter)() const, void (RenderStyle::*setter)(std::optional<Length>), OptionSet<Flags> flags = { })
+        : PropertyWrapperGetter<std::optional<Length>>(property, getter)
+        , m_setter(setter)
+        , m_flags(flags)
+    {
+    }
+
+protected:
+    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation) const override
+    {
+        if (!this->value(from).has_value() || !this->value(to).has_value())
+            return false;
+
+        bool isLengthPercentage = m_flags.contains(Flags::IsLengthPercentage);
+        return canInterpolateLengths(this->value(from).value(), this->value(to).value(), isLengthPercentage);
```

Comment:
And here you can do:

```
return canInterpolateLengths(*this->value(from), *this->value(to), isLengthPercentage);
```

Reply:
The compiler complains about this change, looks like it can convert `std::optional<Length>` to `Length` implicitly. 
```
static bool canInterpolateLengths(const Length& from, const Length& to, bool isLengthPercentage)
```

Reply:
Ah, looks like I missed adding the operator *... It's done now.